src/http/response_parser.hpp

```diff
@@ -48,6 +50,163 @@ class ResponseParser : public NonCopyable, public NonMovable {
     ResponseParserImpl *impl = nullptr;
 };
 
+// Header parsing states (see do_header_internal)
+#define S_NOTHING 0
+#define S_FIELD 1
```

Comment:
I would prefer using enums instead of macros, so you have some compilation check and you can't give to the finite state machine a number as a state!